### PR TITLE
GUI window properties handling

### PR DESCRIPTION
**Pull request by Nadahar**
This replaces the existing implementation for loading and saving the GUI window size and location. In addition to fixing some existing bugs, this implementation stores the information in a separate file (instead of in the configuration file), handles multiple screens, screen insets, preserves the non-maximized size when maximized and more. It also stores any changes immediately instead of at application shutdown, and handles some "tricky" situation like if the screen resolution is changed so that the GUI window is bigger than the screen resolution (relevant when using remote desktop software).

---
This change is [Reviewable](https://reviewable.io/reviews/digitalmediaserver/digitalmediaserver/25)

### CODE REVIEW

File: src/main/java/net/pms/newgui/LooksFrame.java [hunk 1]
```diff
@@ -191,10 +195,14 @@ public static void initializeLookAndFeel() {
 	 * Constructs a <code>DemoFrame</code>, configures the UI,
 	 * and builds the content.
 	 */
-	public LooksFrame(PmsConfiguration configuration) {
+	public LooksFrame(@Nonnull PmsConfiguration configuration, @Nonnull WindowPropertiesConfiguration windowConfiguration) {
+		super(windowConfiguration.getGraphicsConfiguration());
+		if (configuration == null) {
+			throw new IllegalArgumentException("configuration can't be null");
```
Comment (Sami32) [inline src/main/java/net/pms/newgui/LooksFrame.java]: **C**onfiguration

Comment (Nadahar) [reply src/main/java/net/pms/newgui/LooksFrame.java]: @Sami32 It's intentional. Whenever I refer to a particular parameter name, I always use the capitalization of the parameter.

File: src/main/java/net/pms/newgui/components/WindowProperties.java [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+	 * {@link WindowProperties} instance.
+	 *
+	 * @author Nadahar
+	 */
+	@NotThreadSafe
+	public static class WindowPropertiesConfiguration {
+
+		private static final Logger LOGGER = LoggerFactory.getLogger(WindowPropertiesConfiguration.class);
+
+		private static final byte[] MAGIC_BYTES = {(byte) 68, (byte) 71, (byte) 83};
```
Comment (Sami32) [inline src/main/java/net/pms/newgui/components/WindowProperties.java]: Is that related to your propeties stockage format ?

Comment (Nadahar) [reply src/main/java/net/pms/newgui/components/WindowProperties.java]: @Sami32 I don't know what "stockage format" is, but yes - I made my own "magic bytes" to be able to easily recognize the file format.

**Comment by Nadahar** [comment]
Reviewed 3 of 5 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/digitalmediaserver/digitalmediaserver/25)*
